net/gpsd/Portfile

```diff
@@ -227,3 +236,13 @@ description {Include xgps/xgpsspeed X11 clients (dependency-intensive)} {
 if {![variant_isset xgps]} {
     notes "The xgps variant is now needed to get the xgps and xgpsspeed programs."
 }
+
+# Handle the obsolete subport last so that it can override the dependencies.
+
+if {${subport} ne ${name}} {
```

Comment:
2 parts here:
(1) keep the version the same as it was before for the `devel` port; just do a rev-bump for the change (being `replaced_by`) to be engaged.
(2) please just move this code into the subport declaration itself, to be cleaner: as this is the only code being used by the `devel` subport, keep it all together!
Thx!

Reply:
> (1) keep the version the same as it was before for the `devel` port; just do a rev-bump for the change (being `replaced_by`) to be engaged.

Done.  It doesn't seem to mind setting the version/revison twice, which is good because it doesn't require putting the normal version/revision in a conditional.  With tcl one can't necessarily assume that a simple redefinition works as expected.

> (2) please just move this code into the subport declaration itself, to be cleaner: as this is the only code being used by the `devel` subport, keep it all together!

Done.  Originally I thought that making some mention of the subport earlier was useful, but I've now been persuaded otherwise.

If there's a way to eradicate the variants after defining them, it's not documented.  I didn't want to put the variant blocks in a conditional.

I also added a date to the comment, to provide a guideline on when to remove the subport without having to look it up in the git history.  AIUI it's normal practice to keep obsolete ports around for at least a year.

Based on the nature of the changes, I only retested on one machine.

In the typical fashion, the Azure pipelines have completed before the Travis runs have even started. :-)